### PR TITLE
adding custom monster create, edit, and delete

### PR DESCRIPTION
**Pull request by ajones119**
[screen-capture (2).webm](https://github.com/ajones119/BarrysNotesOnBarovia/assets/56804418/afd4ffdb-5750-43a4-bb00-0a189f6a99ed)

Need to clean up UI some but this should include a newer/better way of handling Create/Update/Delete/Read hooks alongside using a typed document. it adds the functionality for making custom monsters, starting the process of moving npc and npc cards onto BaseCharacter and BaseCharacter cards

### CODE REVIEW

File: src/components/Drawer/CreateOrEditMonsterDrawer/index.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+                        <TextInput value={monster?.armorClass} onChange={(value) => setMonster({ ...monster, armorClass: Number(value)})} placeholder='Armor Class' number />
+                    </Grid>
+                    <Grid item xs={4} md={2}>
+                        <TextInput value={monster?.passivePerception} onChange={(value) => setMonster({ ...monster, passivePerception: Number(value)})} placeholder='Passive Percep' number />
+                    </Grid>
+                    <Grid item xs={4} md={1}>
+                        <TextInput value={monster?.abilityScores?.strength} onChange={(value) => {
+                            const scores = {...monster?.abilityScores, strength: Number(value)}
+                            setMonster({...monster, abilityScores: scores})
+                        }} placeholder={`Strength (${Math.floor((monster?.abilityScores?.strength - 10)/2) >= 0 ? "+" : ""}${Math.floor((monster?.abilityScores?.strength - 10)/2)})`} number />
```
Comment (ColtonH) [inline src/components/Drawer/CreateOrEditMonsterDrawer/index.tsx:84]: Random note - shoujld make these a util at some point lol. Its the same calc for every ability score type

**Review by ColtonH — APPROVED** [review]

File: src/components/BaseCharacterThumbCard/Cards/MonsterCard.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import BaseCharacterThumbCard, { BaseCharacterThumbCardProps } from "../BaseCharacterThumbCard";
+import { useDeleteCustomMonster } from "@services/CustomMonstersService";
+
+declare interface MonsterCardProps extends BaseCharacterThumbCardProps {
+
+}
```
[the remark this answers is not in the document]
Comment (ajones119) [reply src/components/BaseCharacterThumbCard/Cards/MonsterCard.tsx:7]: lol totally overlooked removing that

File: src/components/Drawer/CreateOrEditMonsterDrawer/index.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+import { Typography } from "@components/Typography/Typography";
+import SavingThrowSelect from "@components/Selects/SavingThrowSelect";
+import SkillsSelect from "@components/Selects/SkillsSelect";
+import { Button } from "@components/Button/Button";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faXmarkCircle } from "@fortawesome/free-solid-svg-icons";
+import { useCreateCustomMonster, useEditCustomMonster } from "@services/CustomMonstersService";
+import { LoadingButton } from "@components/Button/LoadingButton";
+
+declare interface CreateMonsterDrawerProps extends DrawerProps {
```
[the remark this answers is not in the document]
Comment (ajones119) [reply src/components/Drawer/CreateOrEditMonsterDrawer/index.tsx]: Changed to MonsterDrawer that wraps around BaseCharacterDrawer

File: src/components/Drawer/CreateOrEditMonsterDrawer/index.tsx (hunk 1 above)
Comment (ajones119) [reply src/components/Drawer/CreateOrEditMonsterDrawer/index.tsx]: #truuuueeee

File: src/components/Drawer/CreateOrEditMonsterDrawer/index.tsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    editMonster?: BaseCharacter | null;
+}
+
+const CreateOrEditMonsterDrawer = ({isOpen, onClose = () => {}, editMonster}: CreateMonsterDrawerProps) => {
```
[the remark this answers is not in the document]
Comment (ajones119) [reply src/components/Drawer/CreateOrEditMonsterDrawer/index.tsx]: I like the upsert, but we may have to do that within the app itself. I might keep this for now, and I'll try to think of a good way of handling upserts while I convert the other services away from useServiceButton and to useService

**Pull request merged**